src/client/core/evaluator.ts

```diff
@@ -47,3 +60,113 @@ export const variableReducer = (
   accVariables[id] = evaluateOperation(expression, accVariables)
   return accVariables
 }
+
+/**
+ * Extract all variables from a given mathjs expression
+ * @param expression mathjs expression
+ */
+export const getDependencies = (expression: string): Set<string> => {
+  // Parse the mathjs expression into an expression tree
+  const node = math.parse!(expression)
+
+  // Traverse tree to extract symbols and functions
+  const dependencies = new Set<string>()
+  const functions: string[] = []
+  node.traverse((node) => {
+    if (node.isFunctionNode && node.name) functions.push(node.name)
+    if (node.isSymbolNode && node.name) dependencies.add(node.name)
```

Comment:
could we do `node.isSymbolNode && !node.isFunctionNode && node.name` instead?

Reply:
Wanted to do that at first. But the way the expression tree is structure for say `ifelse(A == B, C, D)` is that there will be two nodes for `ifelse`, one for a `SymbolNode` and one for a `FunctionNode`.